From: Sam Morris <sam@robots.org.uk>
Date: Fri, 20 Jul 2018 14:20:24 +0100
Subject: cmake-3-port

---
 source/iked/CMakeLists.txt | 22 ++++++++--------------
 1 file changed, 8 insertions(+), 14 deletions(-)

diff --git a/source/iked/CMakeLists.txt b/source/iked/CMakeLists.txt
index 2e41586..1ef60a5 100644
--- a/source/iked/CMakeLists.txt
+++ b/source/iked/CMakeLists.txt
@@ -102,27 +102,21 @@ set_source_files_properties(
 	conf.parse.cpp GENERATED,
 	conf.token.cpp GENERATED )
 
-# Custom target for parser
-
-add_custom_target( ConfParser echo "Creating Conf Parser" )
-
 # Custom command for flex
 
 add_custom_command(
-	SOURCE conf.token.ll
-	COMMAND ${PATH_BIN_FLEX}
-	ARGS -d -oconf.token.cpp conf.token.ll
-	TARGET ConfParser
-	OUTPUTS conf.token.cpp )
+	OUTPUT conf.token.cpp
+	COMMAND ${PATH_BIN_FLEX} -d -oconf.token.cpp ${CMAKE_CURRENT_SOURCE_DIR}/conf.token.ll
+	MAIN_DEPENDENCY conf.token.ll
+	)
 
 # Custom command for bison
 
 add_custom_command(
-	SOURCE conf.parse.yy
-	COMMAND ${PATH_BIN_BISON}
-	ARGS -oconf.parse.cpp conf.parse.yy
-	TARGET ConfParser
-	OUTPUTS conf.parse.cpp conf.parse.hpp location.hh position.hh stack.hh )
+	OUTPUT conf.parse.cpp conf.parse.hpp location.hh position.hh stack.hh
+	COMMAND ${PATH_BIN_BISON} -oconf.parse.cpp ${CMAKE_CURRENT_SOURCE_DIR}/conf.parse.yy
+	MAIN_DEPENDENCY conf.parse.yy
+	)
 
 # install support
 
