From: Sam Morris <sam@robots.org.uk>
Date: Fri, 20 Jul 2018 14:20:24 +0100
Subject: cmake-3-port

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8170453..b57501b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,6 +14,7 @@ cmake_minimum_required( VERSION 2.4 )
 if( COMMAND cmake_policy )
 
 	cmake_policy( SET CMP0003 NEW )
+	cmake_policy( SET CMP0050 OLD )
 
 endif( COMMAND cmake_policy )
 
